File: library/src/main/java/com/eggheadgames/inapppayments/IAPManager.java
package com.eggheadgames.inapppayments;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Context;

import com.billing.BillingService;
import com.billing.PurchaseServiceListener;
import com.billing.SubscriptionServiceListener;
import com.billing.amazon.AmazonBillingService;
import com.billing.google.GoogleBillingService2;

import java.util.List;

//Public front-end for IAP functionality. 

public class IAPManager {

    public static final int BUILD_TARGET_GOOGLE = 0;
    public static final int BUILD_TARGET_AMAZON = 1;

    @SuppressLint("StaticFieldLeak")
    private static BillingService billingService;

    public static void build(Context context, int buildTarget, List<String> iapkeys) {
        Context applicationContext = context.getApplicationContext();
        Context contextLocal = applicationContext == null ? context : applicationContext;

        //Build-specific initializations
        if (buildTarget == BUILD_TARGET_GOOGLE) {
            billingService = new GoogleBillingService2(contextLocal, iapkeys);
        } else if (buildTarget == BUILD_TARGET_AMAZON) {
            billingService = new AmazonBillingService(contextLocal, iapkeys);
        }
    }

    public static void init(String key, boolean enableLogging) {
        billingService.init(key);
        billingService.enableDebugLogging(enableLogging);
    }

    public static void addPurchaseListener(PurchaseServiceListener purchaseServiceListener) {
        billingService.addPurchaseListener(purchaseServiceListener);
    }

    public static void removePurchaseListener(PurchaseServiceListener purchaseServiceListener) {
        billingService.removePurchaseListener(purchaseServiceListener);
    }

    public static void addSubscriptionListener(SubscriptionServiceListener subscriptionServiceListener) {
        billingService.addSubscriptionListener(subscriptionServiceListener);
    }

    public static void removeSubscriptionListener(SubscriptionServiceListener subscriptionServiceListener) {
        billingService.removeSubscriptionListener(subscriptionServiceListener);
    }

    public static void buy(Activity activity, String sku, int id) {
        billingService.buy(activity, sku, id);
    }

    public static void subscribe(Activity activity, String sku, int id) {
        billingService.subscribe(activity, sku, id);
    }

    public static void unsubscribe(Activity activity, String sku, int id) {
        billingService.unsubscribe(activity, sku, id);
    }

    public static void destroy() {
        billingService.close();
    }

    public static BillingService getBillingService() {
        return billingService;
    }
}

File: library/src/main/java/com/eggheadgames/inapppayments/IAPManager.kt
package com.eggheadgames.inapppayments

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Context
import com.billing.BillingService
import com.billing.IBillingService
import com.billing.PurchaseServiceListener
import com.billing.SubscriptionServiceListener

//Public front-end for IAP functionality.
object IAPManager {

    @SuppressLint("StaticFieldLeak")
    private var mBillingService: IBillingService? = null

    /**
     * @param context          - application context
     * @param iapKeys          - list of sku for purchases
     * @param subscriptionKeys - list of sku for subscriptions
     */
    @JvmStatic
    fun build(context: Context, iapKeys: List<String>, subscriptionKeys: List<String> = emptyList()) {
        val contextLocal = context.applicationContext ?: context
        mBillingService = BillingService(contextLocal, iapKeys, subscriptionKeys)
    }

    /**
     * Initialize billing service.
     *
     * @param key - key to verify purchase messages. Currently valid only for Google Billing. Leave empty if you want to skip verification
     */
    @JvmStatic
    fun init(key: String? = null, enableLogging: Boolean = false) {
        getBillingService().init(key)
        getBillingService().enableDebugLogging(enableLogging)
    }

    @JvmStatic
    fun addPurchaseListener(purchaseServiceListener: PurchaseServiceListener) {
        getBillingService().addPurchaseListener(purchaseServiceListener)
    }

    @JvmStatic
    fun removePurchaseListener(purchaseServiceListener: PurchaseServiceListener) {
        getBillingService().removePurchaseListener(purchaseServiceListener)
    }

    @JvmStatic
    fun addSubscriptionListener(subscriptionServiceListener: SubscriptionServiceListener) {
        getBillingService().addSubscriptionListener(subscriptionServiceListener)
    }

    @JvmStatic
    fun removeSubscriptionListener(subscriptionServiceListener: SubscriptionServiceListener) {
        getBillingService().removeSubscriptionListener(subscriptionServiceListener)
    }

    @JvmStatic
    fun buy(activity: Activity, sku: String) {
        getBillingService().buy(activity, sku)
    }

    @JvmStatic
    fun subscribe(activity: Activity, sku: String) {
        getBillingService().subscribe(activity, sku)
    }

    @JvmStatic
    fun unsubscribe(activity: Activity, sku: String) {
        getBillingService().unsubscribe(activity, sku)
    }

    @JvmStatic
    fun destroy() {
        getBillingService().close()
    }

    @JvmStatic
    fun getBillingService(): IBillingService {
        return mBillingService ?: let {
            throw RuntimeException("Call IAPManager.build to initialize billing service")
        }
    }
}